src/server/pps/server/s3g_sidecar.go

```diff
@@ -85,6 +91,16 @@ func (a *apiServer) ServeSidecarS3G() {
 		return // break early (nothing to serve via S3 gateway)
 	}
 
+	go func() {
+		for i := 0; i < 2; i++ { // If too many errors, the worker will fail the job
+			err := s.server.ListenAndServe()
+			if err == nil || errors.Is(err, http.ErrServerClosed) {
+				break // server was shutdown/closed
+			}
+			logrus.Errorf("error serving sidecar s3 gateway: %v; strike %d/3", err, i+1)
```

Comment:
I know it was already written this way, but doesn't it only get two strikes?

Reply:
yep that's true. Will update the comment